frontend/src/components/Time/TimeDurationIndicatorComponent.tsx

```diff
@@ -0,0 +1,102 @@
+import * as React from "react";
+import { style } from "typestyle";
+import { Button, Tooltip } from "@patternfly/react-core";
+import { config } from "../../config";
+import { KialiIcon } from "../../config/KialiIcon";
+import {DurationInSeconds, guardTimeRange, TimeRange} from "../../types/Common";
+import { getName, getRefreshIntervalName } from "../../utils/RateIntervals";
+import {KialiAppState} from "../../store/Store";
+import {durationSelector, refreshIntervalSelector, timeRangeSelector} from "../../store/Selectors";
+import {connect} from "react-redux";
+import history, {HistoryManager, URLParam} from "../../app/History";
+import {KialiDispatch} from "../../types/Redux";
+import {bindActionCreators} from "redux";
+import {UserSettingsActions} from "../../actions/UserSettingsActions";
+
+interface Props {
+  isDuration?: boolean;
+  onClick?: () => void;
+  setDuration: (duration: DurationInSeconds) => void;
+  duration: DurationInSeconds;
+  refreshInterval: number;
+  timeRange: TimeRange;
+}
+
+const infoStyle = style({
+  margin: '0px 5px 2px 5px',
+  verticalAlign: '-5px !important'
+});
+
+export class TimeDurationIndicatorComponent extends React.PureComponent<Props> {
+  constructor(props: Props) {
+    super(props);
+    const urlParams = new URLSearchParams(history.location.search);
+    const urlDuration = HistoryManager.getDuration(urlParams);
+    if (urlDuration !== undefined && urlDuration !== props.duration) {
+      props.setDuration(urlDuration);
+    }
+    HistoryManager.setParam(URLParam.DURATION, String(this.props.duration));
```

Comment:
When comparing the old code, this is additional code that I am able to identify.

Can you add a code-comment about why this is needed? I may think that since this is simply an indicator, no changes to the state should be needed. Although, I'm really not sure.

Also, perhaps since this control is dealing with both duration and timeRefresh, maybe similar code is needed for timeRefresh?

Reply:
Comment added. I've seen that timeRefresh doesn't need it. I think this is because it is shared, but the Duration or the TimeRange are different controls.